dvc/config.py

```diff
@@ -273,11 +277,16 @@ class Config(object):  # pylint: disable=too-many-instance-attributes
         Optional(SECTION_STATE_ROW_CLEANUP_QUOTA): And(Use(int), is_percent),
     }
 
+    SECTION_OAUTH2 = "oauth2"
```

Comment:
Could you please explain why we need a separate section instead of just using gdrive remote section?

Reply:
This section contains the information about the client environment, ss. is it running on a local machine when we could run the flow with local webserver + webbrowser, or is it a server accessed by SSH which couldn't run the browser and pass the callback to local webserver (so we ask the user to open the page, authorize DVC in Google OAuth dialog in their browser, and copy the code from there.

This information is not related to the remote. And also should (better) be defined in the global (not per-repo) config.

Reply:
if it's a global OAuth setting, let's make it a `core` section option? I also don't see immediately a reason to have a whole global section.